.github/workflows/compound-java-check.yml

```
@@ -0,0 +1,147 @@
+      cisa-domain:
+        required: false
+        description: 'Cybersecurity and Infrastructure Security Agency domain'
+      cisa-user:
+        required: false
+        description: 'Cybersecurity and Infrastructure Security Agency user'
+      cisa-password:
+        required: false
+        description: 'Cybersecurity and Infrastructure Security Agency password'
+    outputs:
+      owasp-scan-status:
+        value: ${{ jobs.calculate-statuses.outputs.owasp-check-status }}
+      license-check-status:
+        value: ${{ jobs.calculate-statuses.outputs.license-check-status }}
+      integration-test-status:
+        value: ${{ jobs.calculate-statuses.outputs.integration-test-status }}
+      aggregated-status:
+        description: "Aggregated status"
+        value: ${{ jobs.calculate-statuses.outputs.aggregated-status }}
+
+jobs:
+  owasp-scan:
+    name: 'OWASP scan'
+    if: inputs.scanner-enabled
+    uses: th2-net/.github/.github/workflows/owasp-gradle-scan.yml@integration-tests # FIXME: switch to main branch
+    with:
+      runsOn: ${{ inputs.runs-on }}
+      javaVersion: ${{ inputs.java-version }}
+      javaVendor: ${{ inputs.java-vendor }}
+    secrets:
+        nvd-api-key: ${{ secrets.nvd-api-key }}
+        cisa-domain: ${{ secrets.cisa-domain }}
+        cisa-user: ${{ secrets.cisa-user }}
+        cisa-password: ${{ secrets.cisa-password }}
+
+  license-check:
+    name: 'License check'
+    if: inputs.scanner-enabled
+    uses: th2-net/.github/.github/workflows/license_check.yml@main
+    with:
+      runsOn: ${{ inputs.runs-on }}
+      version: ${{ inputs.app-version }}
+
+  integration-test:
+    name: 'Integration tests'
+    if: inputs.integration-test-enabled
+    strategy:
+      matrix:
+       integration-test-project: ${{ fromJson(inputs.integration-test-projects) }}
+    uses: th2-net/.github/.github/workflows/java-integration-test.yml@integration-tests # FIXME: switch to main branch
+    with:
+      runs-on: ${{ inputs.runs-on }}
+      java-version: ${{ inputs.java-version }}
+      java-vendor: ${{ inputs.java-vendor }}
+      project-path: ${{ matrix.integration-test-project }}
+      integration-test-task: ${{ inputs.integration-test-task }}
+
+  calculate-statuses:
+    name: 'Calculate statuses'
+    runs-on: ${{ inputs.runs-on }}
+    if: always()
+    needs: 
+      - owasp-scan
+      - license-check
+      - integration-test
+    outputs:
+      owasp-check-status: ${{ steps.owasp-check-status.outputs.value }}
+      license-check-status: ${{ steps.license-check-status.outputs.value }}
+      integration-test-status: ${{ steps.integration-test-status.outputs.value }}
+      aggregated-status: ${{ steps.aggregated-status.outputs.value }}
+    steps:
+      - name: Prepare OWASP check status
+        id: owasp-check-status
+        run: echo "value=${{ inputs.strict-scanner && needs.owasp-scan.result == 'success' || !inputs.strict-scanner }}" >> $GITHUB_OUTPUT
```

suggestion: I think we could simplify the expression here
```suggestion
        run: echo "value=${{ !inputs.strict-scanner || needs.owasp-scan.result == 'success' }}" >> $GITHUB_OUTPUT
```